### ISSUE TITLE
[10.0] NameError: global name 'line' is not defined in PoS onchange

### ISSUE THREAD

**Issue by aisopuro** [odoo/odoo#19112]
Impacted versions: 10.0 

Steps to reproduce:

We received a logging of a traceback to our error tracking service (below), but looking at the code, this would seem to be able to do it:
1. Create a PoS order
2. Create a line on the PoS order
3. Set a fiscal position on the PoS order
4. Change the product on the line

Current behavior:

An exception is raised (again, traceback below).

Expected behavior:

The user should be able to change the product without an exception being raised.

Comments:

This actually seems pretty clear to me: the variable name `line` isn't mentioned anywhere else in the method.

The latest version of the file at time of writing is this one: https://github.com/odoo/odoo/blob/f69655829a1b67b4b573a8f8b661d29ed9f30e16/addons/point_of_sale/models/pos_order.py

Traceback:
```
NameError: global name 'line' is not defined
  File "odoo/http.py", line 640, in _handle_exception
    return super(JsonRequest, self)._handle_exception(exception)
  File "odoo/http.py", line 677, in dispatch
    result = self._call_function(**self.params)
  File "odoo/http.py", line 333, in _call_function
    return checked_call(self.db, *args, **kwargs)
  File "odoo/service/model.py", line 101, in wrapper
    return f(dbname, *args, **kwargs)
  File "odoo/http.py", line 326, in checked_call
    result = self.endpoint(*a, **kw)
  File "odoo/http.py", line 935, in __call__
    return self.method(*args, **kw)
  File "odoo/http.py", line 506, in response_wrap
    response = f(*args, **kw)
  File "odoo/addons/web/controllers/main.py", line 885, in call_kw
    return self._call_kw(model, method, args, kwargs)
  File "odoo/addons/web/controllers/main.py", line 877, in _call_kw
    return call_kw(request.env[model], method, args, kwargs)
  File "odoo/api.py", line 681, in call_kw
    return call_kw_multi(method, model, args, kwargs)
  File "odoo/api.py", line 672, in call_kw_multi
    result = method(recs, *args, **kwargs)
  File "odoo/models.py", line 5497, in onchange
    record._onchange_eval(name, field_onchange[name], result)
  File "odoo/models.py", line 5395, in _onchange_eval
    method_res = method(self)
  File "odoo/addons/point_of_sale/models/pos_order.py", line 871, in _onchange_product_id
    tax_ids_after_fiscal_position = fpos.map_tax(self.tax_ids, line.product_id, line.order_id.partner_id) if fpos else self.tax_ids
```

**Comment by aisopuro** [comment]
It seems to me that something may have gone wrong during a merge: it seems that 657e3289032d8ce6466019a36f7abf1be49df5cc merged in this new logic for handling the fiscal position (specifically [this line](https://github.com/odoo/odoo/commit/657e3289032d8ce6466019a36f7abf1be49df5cc#diff-5eeb7d6a1a488dff039e5cc3300730e1R871)). Does @KangOl know what may have gone wrong?

**Comment by KangOl** [comment]
Yes, I know. I screw up the forward-port. Fixed with 38c77a521087405d8b42f24c882f19ada0387821
